### ISSUE TITLE
Initialisation callback is never called when Google ad ID is not available

### ISSUE THREAD

**Issue by nchuzh** [BranchMetrics/android-branch-deep-linking-attribution#1148]
### Describe the bug

BranchUniversalReferralInitListener.onInitFinished  is never called if retrieving Google advertising ID fails.
Our app relies on the initialisation callback to make a decision where to navigate on the app launch and the issue leads to the app freeze.

Here are the snippets of Branch library code.
The bug in Branch SDK appears to be specifically for Google advertising ID,  when google  service is not available it returns null and null is being ignored with the coroutine job
![Screenshot 2023-11-22 at 11 41 08 (1)](https://github.com/BranchMetrics/android-branch-deep-linking-attribution/assets/10884225/d265c3db-b870-4be9-8344-ad60fe4e4179)

on the other hand, Amazon fire advertising ID is handled correctly with finally  for null value.
![Screenshot 2023-11-22 at 11 48 41 (1)](https://github.com/BranchMetrics/android-branch-deep-linking-attribution/assets/10884225/ed06e960-fcea-471d-a2d4-0d83a297f89b)

### Steps to reproduce

1. Set initialisation code
```
      Branch.sessionBuilder(activity)
            .withCallback { 
            }
            .init()
```
2. Install the app on an emulator without Google services
3. Observe `GooglePlayServicesNotAvailableException` in the logs and the callback being never called

### Expected behavior

BranchUniversalReferralInitListener.onInitFinished must be called with either error or success

### SDK Version

5.7.2

### Make and Model

emulator

### OS

API34

**Comment by gdeluna-branch** [comment]
Thanks for raising this, we should get to it this sprint.

**Comment by gdeluna-branch** [comment]
Should be fixed in 5.8.0. Please reopen if you can still repro. Thanks!
